viewing_party/party.py

```diff
@@ -0,0 +1,150 @@
+#wave 1
+def create_movie(movie_title, genre, rating):
+    if movie_title and genre and rating:
+        movie_dict = {
+            "title" : movie_title,
+            "genre" : genre,
+            "rating" : rating,
+        }
+        return movie_dict
+    else:
+        return None
+
+#wave 2
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie in user_data["watchlist"]:
+        print("This movie is already in the watchlist!")
+    else:     
+        user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            break
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    all_ratings = 0
+    if len(user_data["watched"]) == 0: 
+        print("We haven't started watching yet!")
+        return 0
+    for movie in user_data["watched"]:
+        all_ratings += movie["rating"]
+    return all_ratings / len(user_data["watched"])
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    if len(user_data["watched"]) == 0: 
+        print("We haven't started watching yet!")
+        return None
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_dict:
+            genre_dict[movie["genre"]] = 1
+        else:
+            genre_dict[movie["genre"]] += 1
+
+    genre_tally = 0
+    fave_genre = ""
+    for genre, count in genre_dict.items():
+        if count > genre_tally:
+            genre_tally = count
+            fave_genre = genre
+    return fave_genre
+
+
+#wave 3
+def get_unique_watched(user_data):
+    only_friends_watched =[]
+    my_unique_watched = []
+    for watched in user_data["friends"]:
+        for movie in watched["watched"]:
+            only_friends_watched.append(movie["title"])
+    for movie in user_data["watched"]:
+        if movie["title"] not in only_friends_watched:
+            my_unique_watched.append(movie)
+    return my_unique_watched
```

Comment:
we will definitely be seeing these for loops again and again! maybe making these helper functions would help in the future! When a project is this small, it may not matter as much, but it could say you from creating a repeat for loop here and there.